src/Generator/EnumGenerator/Cases/BackedCases.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator\Cases;
+
+use Laminas\Code\Generator\EnumGenerator\CasesInterface;
+
+final class BackedCases implements CasesInterface
+{
+    private string $type;
+
+    /** @var list<string> */
+    private array $cases;
+
+    /**
+     * @param list<string> $cases
+     */
+    private function __construct(string $type, array $cases)
+    {
+        $this->type  = $type;
+        $this->cases = $cases;
+    }
+
+    public function getBackedType(): ?string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getCases(): array
+    {
+        return $this->cases;
+    }
+
+    /**
+     * @param array<string, string|int> $backedCases
+     */
+    public static function fromCasesWithType(array $backedCases, string $type): self
+    {
+        $cases = [];
+        foreach ($backedCases as $case => $value) {
+            if ($type === 'string') {
+                $value = '\'' . $value . '\'';
```

Comment:
I'd go with either:

```php
$value = "'" . $value . "'";
```

or:

```php
$value = sprintf("'%s'", $value);
```

to make it more clear that what you're doing is surrounding a string value with single quotes.